src/components/LmsConfigurations/BlackboardIntegrationConfigForm.jsx

```diff
@@ -0,0 +1,284 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import {
+  ValidationFormGroup, Input, StatefulButton, Icon,
+} from '@edx/paragon';
+import { snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import StatusAlert from '../StatusAlert';
+import NewRelicService from '../../data/services/NewRelicService';
+import SUBMIT_STATES from '../../data/constants/formSubmissions';
+
+export const REQUIRED_BLACKBOARD_CONFIG_FIELDS = [
+  'blackboardBaseUrl',
+  'clientId',
+  'clientSecret',
+];
+
+class BlackboardIntegrationConfigForm extends React.Component {
```

Comment:
Same as I wrote to Talia; do not refactor this component, but for the future: we're moving away from React class components in favor of functional components. All new React code we write should be functional components unless there's a compelling reason not to (In this case, it's very similar to another component.)